trivy-scan-image/action.yml

```diff
@@ -0,0 +1,77 @@
+# Required permissions:
+#   contents: read
+#   security-events: write
+
+name: Run Trivy vulnerability scanner
+description: Run Trivy vulnerability scanner on an image and upload the results
+
+inputs:
+  image-tag:
+    description: Full image ref (e.g. `ghcr.io/flagsmith/flagsmith:main`).
+    required: true
+  category:
+    description: CodeQL category (usually image name, e.g. `flagsmith-api`).
+    required: true
+  query:
+    description: CodeQL query (used for report URLs).
+    required: false
+  trivy-username:
+    description: TRIVY_USERNAME value.
+    required: false
+  trivy-password:
+    description: TRIVY_PASSWORD value.
+    required: false
+  trivy-db-repository:
+    description: TRIVY_DB_REPOSITORY value.
+    required: false
+  trivy-java-db-repository:
+    description: TRIVY_JAVA_DB_REPOSITORY value.
+    required: false
+
+outputs:
+  scan-results-url:
+    description: URL to scan results in Github Security tab
+    value: ${{ steps.scan-results-url.outputs.scan-results-url }}
+  scan-outcome:
+    description: Scan outcome
+    value: ${{ steps.trivy.outcome }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Checkout trivy.yaml
+      uses: actions/checkout@v4
+      with:
+        sparse-checkout: |
+          trivy.yaml
+          .github*
+        sparse-checkout-cone-mode: false
+
+    - name: Run Trivy on ${{ inputs.image-tag }}
+      id: trivy
+      uses: aquasecurity/trivy-action@master
+      with:
+        image-ref: ${{ inputs.image-tag }}
+        format: sarif
+        output: trivy-results.sarif
+      env:
+        TRIVY_USERNAME: ${{ inputs.trivy-username }}
+        TRIVY_PASSWORD: ${{ inputs.trivy-password }}
+        TRIVY_DB_REPOSITORY: ${{ inputs.trivy-db-repository }}
+        TRIVY_JAVA_DB_REPOSITORY: ${{ inputs.trivy-java-db-repository }}
+
+    - name: Upload Trivy scan results to GitHub Security tab
+      uses: github/codeql-action/upload-sarif@v3
+      if: success() || failure()
+      with:
+        category: ${{ inputs.category }}
+        sarif_file: trivy-results.sarif
+
+    - name: Render scan results URL
+      if: success() || failure()
+      id: scan-results-url
+      run: >
+        echo scan-results-url=${{ format('{0}/{1}/security/code-scanning?query=is:open+tool:Trivy+{2}',
+        github.server_url, github.repository, inputs.query) }} >> $GITHUB_OUTPUT
```

Comment:
Are we sure `github.repository` is going to reference the calling workflow and not the `Flagsmith/actions` one?

Reply:
Good question. I'll test it out. Not sure why I hadn't already tbh.

Reply:
Based on [this workflow run](https://github.com/Flagsmith/flagsmith/actions/runs/11555844042/job/32162204125) in the core repository, and specifically, the following screenshot, I think we're good here. 


<img width="1031" alt="image" src="https://github.com/user-attachments/assets/93b7b707-064b-4b17-9db7-038c32c2c39e">


That being said, I was a little confused by [this](https://github.com/Flagsmith/flagsmith/pull/4766/checks?check_run_id=32162314762) but it does seem like other PRs report similar warnings so I don't think it's related to this PR specifically.